src/mds/MDCache.cc

```diff
@@ -8424,16 +8424,25 @@ int MDCache::path_traverse(MDRequestRef& mdr, MDSContextFactory& cf,
 	    depth == path.depth() - 1) {
 	  ceph_assert(dn->is_auth());
 	  if (depth > 0 || !mdr->lock_cache) {
-	    lov.add_wrlock(&cur->filelock);
+            if (xlock_dentry_parent) {
+              lov.add_xlock(&cur->filelock);
```

Comment:
This won't be appropriate I think. The `filelock` is deliberately configured to allow multiple writers (yes, even and especially for directory inodes) otherwise we will serialize every file create. (Please test this to confirm.)

What appears to be missing is that a wrlock on the `filelock` is not correctly revoking (or waiting for the revoke of) `Fx` on the directory inode. No?

Reply:
> This won't be appropriate I think. The `filelock` is deliberately configured to allow multiple writers (yes, even and especially for directory inodes) otherwise we will serialize every file create. (Please test this to confirm.)
> 

Yeah, this really will serialize the creates.

> What appears to be missing is that a wrlock on the `filelock` is not correctly revoking (or waiting for the revoke of) `Fx` on the directory inode. No?

Yeah, correct. Maybe we can fix it by revoking the `Fx` when acquiring the `wrlock` ?

Reply:
Tested the new patch and it worked well.